src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -470,6 +508,32 @@
            netscape-comment-value)
           (utils/create-ca-extensions ca-name ca-serial ca-public-key))))
 
+(schema/defn read-compile-masters
+    "Returns a list of compile master serials from the specified file organized as one item per line."
+    [cm-file :- schema/Str]
+    (line-seq (io/reader cm-file)))
+
+(schema/defn generate-compile-master-serials
+  "Given a list of compile master nodes it will create a file containing 
+   serial numbers of their certificates (listed on separate lines).
+   Compile masters is expected have at least one entry (MoM)"
+   [cm-nodes :- [schema/Str]
+    compile-masters-serials-file :- schema/Str
+    signeddir :- schema/Str]
+
+   (log/debug (format "%s\n%s" (i18n/trs "generate-compile-master invoked ") compile-masters-serials-file))
```

Comment:
This is probably too high of detail for logging, or if we really want to keep it in it would be a `trace`, I think.